### PR TITLE
[Requirements] Bumping fsspec and dependencies to 2023.9.0

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
[ML-5219](https://jira.iguazeng.com/browse/ML-5219)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -30,7 +30,9 @@ mergedeep~=1.3
 v3io-frames~=0.10.7
 semver~=3.0
 dependency-injector~=4.41
-fsspec>=2023.1,<2023.7
+# in version 2023.9.2 fsspec canceled "auto_mkdir" in open() function
+# we required a future solution
```
Comment (assaf758) [inline requirements.txt:34]:
```suggestion
# upper limit due to version 2023.9.2 fsspec removing the "auto_mkdir" in the open() function. TODO to address this differently
```
also please raise a Jira on this

File: extras-requirements.txt [hunk 2]
```diff
@@ -8,22 +8,22 @@
 # in setup.py so that we'll be able to copy and install this in the layer with all other requirements making the last
 # layer (which is most commonly being re-built) as thin as possible
 # we have a test test_extras_requirement_file_aligned to verify this file is aligned to setup.py
-boto3>=1.24.59,<1.27
-aiobotocore>=2.4.2,<2.6
-s3fs>=2023.1,<2023.7
+boto3>=1.28.0,<1.29.0
+aiobotocore>=2.5.0,<2.8
+s3fs==2023.9.0
```
Comment (assaf758) [inline extras-requirements.txt:13]: please add a comment where this specific version came from.

**Review by assaf758 — CHANGES_REQUESTED** [review]

**Review by theSaarco — APPROVED** [review]
LGTM. We need to be aware of the newly pinned requirements and make sure we update them per need or periodically (`fsspec` and related packages tend to get monthly updates).

File: requirements.txt (hunk 1 above)
Comment (tomerm-iguazio) [reply requirements.txt]: https://jira.iguazeng.com/browse/ML-5235

**Review by assaf758 — APPROVED** [review]

**Review by liranbg — APPROVED** [review]

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# >=1.26.9, <1.27 from botocore 1.19.28 inside boto3 1.16.28 inside nuclio-jupyter 0.8.8
+# >=1.28.0,<1.29.0 from botocore 1.28.64 inside boto3 1.16.28 inside nuclio-jupyter 0.9.13
```
Comment (alonmr) [inline requirements.txt:1]: comment here is not matching the urlib3 versions below

**Pull request merged**